src/solvers/dg/2d/dg.jl

```diff
@@ -1629,32 +1642,137 @@ Calculate the finite volume fluxes inside the elements.
 - `u::AbstractArray{T} where T<:Real`
 - `element_id::Integer`
 """
-@inline function calcflux_fv!(fstar1, fstar2, u, element_id, dg::Dg2D)
+@inline function calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u, element_id, dg::Dg2D, nonconservative_terms::Val{false})
   @unpack surface_flux_function = dg
 
-  fstar1[:, 1,            :] .= zero(eltype(fstar1))
-  fstar1[:, nnodes(dg)+1, :] .= zero(eltype(fstar1))
+  fstar1_L[:, 1,            :] .= zero(eltype(fstar1_L))
```

Comment:
For something like this with array slicing in Julia I believe it will be faster if you use `@views fstar1_L[:, 1,            :] .= zero(eltype(fstar1_L))`

Reply:
Not in this case. From the [Julia manual](https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-views) (highlight is mine):
> In Julia, an array "slice" expression like `array[1:5, :]` creates a copy of that data (**except on the left-hand side of an assignment**, where `array[1:5, :] = ...` assigns in-place to that portion of `array`).

Thus, if you are assigning to a slice, no copy is created (it would defeat the purpose, I guess) and thus no `view` should be necessary if the slice is only on the left-hand side.

Reply:
Correct:
```julia
julia> using BenchmarkTools, StaticArrays

julia> fstar1 = @MArray rand(4, 5, 5)

julia> foo!(fstar1) = fstar1[:, 1, :] .= zero(eltype(fstar1)

julia> bar!(fstar1) = @views fstar1[:, 1, :] .= zero(eltype(fstar1))

julia> @benchmark foo!($(fstar1))
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     25.017 ns (0.00% GC)
  median time:      25.022 ns (0.00% GC)
  mean time:        25.192 ns (0.00% GC)
  maximum time:     163.165 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     996

julia> @benchmark bar!($(fstar1))
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     25.016 ns (0.00% GC)
  median time:      25.021 ns (0.00% GC)
  mean time:        25.148 ns (0.00% GC)
  maximum time:     74.674 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     996
```

Reply:
Wait, so I am confused again. We should use @views then because it is faster? Or do I misinterpret what is going on here

Reply:
That0s also how I see it. I just added the @views calls in 545d0415ef0abe9e648a9daf1588437a0d023cb4

Reply:
No. Don't use `@views` when you have a slice only on the left-hand side of an assignment. Consider using `view`s when you have slices on the right-hand side.

Reply:
And please, try to quote code, in particular if you use macros. Otherwise, you ping some random GitHub user (such as the user `views` here).